meteor/server/api/ingest/commit.ts

```diff
@@ -260,14 +266,20 @@ export async function CommitIngestOperation(
 							if (segment?.document.isHidden) {
 								if (!canRemoveSegment(currentPartInstance, nextPartInstance, segmentId)) {
 									logger.warn(`Cannot hide live segment ${segmentId}, it will be orphaned`)
+									const oldSegment = Segments.findOne(segmentId, {
```

Comment:
The purpose of this is to update the cache?

Reply:
Yes. I don't entirely like that, but this is how it goes: the changes are happening on the cache when the updates to segments are being processed by the blueprints, but in the commit step, we need to revert some. In our case the minishelf can be independent from the segment being hidden or not. If the user floated the on-air story in iNews, both minishelf and the segment timeline will be hidden by the blueprints, and that change goes to the cache, so while we're preventing the segment from hiding we also want to preserve the minishelf by reverting that change.